src/layers/ruby.rs

```diff
@@ -0,0 +1,91 @@
+use std::path::Path;
+
+use crate::util;
+use crate::util::UrlError;
+
+use tempfile::NamedTempFile;
+
+use crate::{RubyBuildpack, RubyBuildpackError};
+use libcnb::build::BuildContext;
+use libcnb::data::layer_content_metadata::LayerTypes;
+use libcnb::generic::GenericMetadata;
+use libcnb::layer::{Layer, LayerResult, LayerResultBuilder};
+
+use crate::RubyVersion;
+use libcnb::data::buildpack::StackId;
+
+pub struct RubyLayer {
+    pub version: RubyVersion,
+}
+
+use url::Url;
+
+impl Layer for RubyLayer {
+    type Buildpack = RubyBuildpack;
+    type Metadata = GenericMetadata;
+
+    fn types(&self) -> LayerTypes {
+        LayerTypes {
+            build: true,
+            launch: true,
+            cache: true,
+        }
+    }
+
+    fn create(
+        &self,
+        context: &BuildContext<Self::Buildpack>,
+        layer_path: &Path,
+    ) -> Result<LayerResult<Self::Metadata>, RubyBuildpackError> {
+        let version = match &self.version {
+            RubyVersion::Explicit(v) => v.clone(),
+            RubyVersion::Default => String::from("3.0.3"),
+        };
+
+        println!("---> Download and extracting Ruby {}", &version);
+
+        let tmp_ruby_tgz =
+            NamedTempFile::new().map_err(RubyBuildpackError::CouldNotCreateTemporaryFile)?;
+
+        let url = RubyLayer::download_url(&context.stack_id, &version)
+            .map_err(RubyBuildpackError::UrlParseError)?;
+
+        util::download(url.as_ref(), tmp_ruby_tgz.path())
+            .map_err(RubyBuildpackError::RubyDownloadError)?;
+
+        util::untar(tmp_ruby_tgz.path(), &layer_path)
+            .map_err(RubyBuildpackError::RubyUntarError)?;
+
+        LayerResultBuilder::new(GenericMetadata::default()).build()
+    }
+}
+
+impl RubyLayer {
+    fn download_url(stack: &StackId, version: impl std::fmt::Display) -> Result<Url, UrlError> {
+        let filename = format!("ruby-{}.tgz", version);
+        let base = "https://heroku-buildpack-ruby.s3.amazonaws.com";
```

Comment:
Using region-specific domain for parity with heroku/heroku-buildpack-ruby/pull/1311

```suggestion
        let base = "https://heroku-buildpack-ruby.s3.us-east-1.amazonaws.com";
```